modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaServiceReferenceAnnotationCheck.java

```diff
@@ -20,19 +20,26 @@
 /**
  * @author Tamyris Bernardo
  */
-public class UpgradeJavaServiceReferenceAnnotationCheck extends BaseFileCheck {
+public class UpgradeJavaServiceReferenceAnnotationCheck
+	extends BaseUpgradeCheck {
 
 	@Override
-	protected String doProcess(
+	protected String afterFormat(
+		String fileName, String absolutePath, String content,
+		String newContent) {
+
+		return StringUtil.replace(
+			newContent,
+			"import com.liferay.portal.spring.extender.service." +
+				"ServiceReference;",
+			"import org.osgi.service.component.annotations.Reference;");
```

Comment:
Do you think adding a dedicated method (e.g. `getReplacedImports()` and `replaceImports()`) would be beneficial for cases like this? That way, the code can just look like this and we would not have to override `afterFormat()`?

```java
@Override
protected Map<String, String> getReplacedmports() {
    return HashMapBuilder.put(
        "com.liferay.portal.spring.extender.service.ServiceReference",
        "org.osgi.service.component.annotations.Reference"
    ).build();
}
```

And in `BaseUpgradeCheck` we would have a method called `replaceImports()`:

```java
protected String afterFormat(...) {
    replaceImports(..., getReplacedImports());
    addNewImports(getNewImports());
    ...
}
```

Reply:
So, I thought about it. I didn't get to do it because by doing so we exchanged one override for another, and the number of lines didn't decrease. So I preferred not to do it. Besides, many classes will override the afterFormat anyway to apply a specific logic. Do you think it is worth doing anyway?

Reply:
It could be worth doing. My worry is that checks will constantly override `afterFormat()`, which is usually a sign that `afterFormat` is not handling enough of the logic. That is something that should be avoided. Though, I am speaking hypothetically. The way `afterFormat()` is now might be good enough for the future.

For now, we can just keep this idea in mind for the future.